create-convex-hull-around-points/src/main/java/com/esri/arcgismaps/sample/createconvexhullaroundpoints/MainActivity.kt

```diff
@@ -0,0 +1,215 @@
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // add point and convex hull graphics to the graphics overlay
+        graphicsOverlay.graphics.addAll(listOf(pointGraphic, convexHullGraphic))
+
+        // create and add a map with topographic basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISTopographic).apply {
+            // set a default initial point and scale
+            initialViewpoint = Viewpoint(Point(34.77, -10.24), 20e7)
+        }
+
+        // configure map view assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        lifecycleScope.launch {
+            // if the map load fails show the error and return
+            map.load().onFailure {
+                showError("Error loading map")
+                return@launch
+            }
+            // capture and collect when the user taps on the screen
+            mapView.onSingleTapConfirmed.collect { event ->
+                event.mapPoint?.let { point ->
+                    addMapPoint(point)
+                }
+            }
+        }
+
+        // add a click listener to create a convex hull
+        createButton.setOnClickListener {
+            // check if the pointGraphic's geometry is not null
+            pointGraphic.geometry?.let { geometry ->
+                createConvexHull(geometry)
+            }
+        }
+
+        // add a click listener to reset the map
+        resetButton.setOnClickListener {
+            resetMap()
+        }
+    }
+
+    /**
+     * Adds the [point] to the map drawn as a Multipoint geometry
+     */
+    private fun addMapPoint(point: Point) {
+        // add the new point to the points list
+        inputPoints.add(point)
+        // recreate the graphics geometry representing the input points
+        pointGraphic.geometry = Multipoint(inputPoints)
+        // enable all the action buttons, since we have at least one point drawn
+        createButton.isEnabled = true
+        resetButton.isEnabled = true
+    }
+
+    /**
+     * Creates and draws a convex hull graphic on the map using [pointGeometry] points
+     */
+    private fun createConvexHull(pointGeometry: Geometry) {
+        // normalize the geometry for panning beyond the meridian
+        val normalizedPointGeometry = GeometryEngine.normalizeCentralMeridian(pointGeometry)
+        // check to proceed if the resulting geometry is not null
+        if (normalizedPointGeometry == null) {
+            // if it is null, show the error and return
+            showError("Error normalizing point geometry")
+            return
+        }
+        // create a convex hull from the points and proceed if it's not null
+        GeometryEngine.convexHull(normalizedPointGeometry)?.let { convexHullGeometry ->
```

Comment:
```suggestion
        // create a convex hull from the points and proceed if it's not null
        val convexHullGeometry = GeometryEngine.convexHull(normalizedPointGeometry)
            ?: return showError("Error creating convex hull")
```
The elvis operator was making it a little hard to follow. Maybe just get the value and return if null right then.

Reply:
I thought `?.<scope function> {} ?: ` was a generally accepted pattern? I did see some other samples using a similar pattern.

Reply:
Yes, that's true, but it would be best to use it for single-assignment operations. Comment by Changan regarding the Elvis operator: [Link](https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/62#discussion_r1105144591)

Reply:
I think in that PR, Changan was referring to the elvis operator branching into a `when` statement instead of handling a nullability clause. 

I've refactored it to a traditional `if` clause.